### PR TITLE
S3 Block Changes.

### PR DESCRIPTION
**Pull request by Gervwyk**
### What are the changes and their implications?
- `AwsS3PresignedPostPolicy` to include `fields` property. Useful for setting `Content-Type` etc.
- Refactor `S3UploadPhoto` and `S3UploadButton`.
- Add `singleFile` and `maxCount` to `S3UploadPhoto` and `S3UploadButton` blocks.
- Add `S3UploadDragger` and `S3Download` blocks.

## Checklist

- [x] Pull request is made to the "develop" branch
- [ ] Tests added
- [x] Documentation added/updated
- [x] Code has been formatted with Prettier
- [x] Edits from maintainers are allowed

### CODE REVIEW

File: packages/docs/blocks/display/S3Download.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    block_type: S3Download
+    value_type: object
+    category: display
+    schema: ../plugins/plugins/plugin-aws/src/blocks/S3Download/schema.json
+    filePath: blocks/display/S3Download.yaml
+    description_content: |
+      The `S3Download` block allows a user to to download a files from AWS S3.
+
+      The `s3GetPolicyRequestId` property is required, and should be the `id` of a [`AwsS3PresignedGetPolicy`](/AWSS3) request. This request is called by the block when the user downloads a file, to create a policy that allows a file to be downloaded to AWS S3.
+
```
Comment (SamTolmay) [inline packages/docs/blocks/display/S3Download.yaml:27]: downloaded from AWS

File: packages/plugins/plugins/plugin-aws/src/blocks/S3Download/S3Download.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+S3Download.defaultProps = blockDefaultProps;
+S3Download.meta = {
+  category: 'display',
+  icons: [],
+  styles: ['blocks/S3Download/style.less'],
+};
+
+export default S3Download;
+
+// Block
```
Comment (SamTolmay) [inline packages/plugins/plugins/plugin-aws/src/blocks/S3Download/S3Download.js:63]: Do we want this here?

File: packages/plugins/plugins/plugin-aws/src/blocks/S3UploadDragger/schema.json [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      },
+      "showUploadList": {
+        "type": "boolean",
+        "default": true,
+        "description": "Whether to show default upload list."
+      },
+      "singleFile": {
+        "type": "boolean",
+        "default": false,
+        "description": "Only allow a single file to be uploaded. Only one file can be selected in the prompt and the upload button is disabled after a file is uploaded."
```
Comment (SamTolmay) [inline packages/plugins/plugins/plugin-aws/src/blocks/S3UploadDragger/schema.json:57]: Is this description correct? Doesn't match the comment in the code?

File: packages/plugins/plugins/plugin-aws/src/blocks/S3UploadPhoto/S3UploadPhoto.js [hunk 4]
```diff
@@ -122,23 +44,30 @@ const S3UploadPhoto = ({ blockId, components: { Icon }, events, methods, propert
...
+  }, [value]);
   return (
     <Upload
       listType="picture-card"
       className="avatar-uploader"
       accept="image/*"
-      customRequest={customRequest}
-      maxCount={1}
+      customRequest={s3UploadRequest}
+      maxCount={properties.maxCount}
```
Comment (SamTolmay) [inline packages/plugins/plugins/plugin-aws/src/blocks/S3UploadPhoto/S3UploadPhoto.js:58]: Please add changeset for this new property. I also don't see it in the schema.json

File: packages/plugins/plugins/plugin-aws/src/connections/AwsS3Bucket/AwsS3PresignedPostPolicy/AwsS3PresignedPostPolicy.js [hunk 5]
```diff
@@ -16,14 +16,15 @@
...
 
 function AwsS3PresignedPostPolicy({ request, connection }) {
   const { accessKeyId, secretAccessKey, region, bucket } = connection;
-  const { acl, conditions, expires, key } = request;
+  const { acl, conditions, expires, key, fields = {} } = request;
   const params = {
     Bucket: bucket,
     Fields: {
-      key,
+      key: key,
```
Comment (SamTolmay) [inline packages/plugins/plugins/plugin-aws/src/connections/AwsS3Bucket/AwsS3PresignedPostPolicy/AwsS3PresignedPostPolicy.js:27]: Is this here for a specific reason? If so then I think it needs a comment so it doesn't get changed back

File: packages/docs/blocks/display/S3Download.yaml [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    value_type: object
+    category: display
+    schema: ../plugins/plugins/plugin-aws/src/blocks/S3Download/schema.json
+    filePath: blocks/display/S3Download.yaml
+    description_content: |
+      The `S3Download` block allows a user to download a files from AWS S3.
+
+      The `s3GetPolicyRequestId` property is required, and should be the `id` of a [`AwsS3PresignedGetPolicy`](/AWSS3) request. This request is called by the block when the user downloads a file, to create a policy that allows a file to be downloaded from AWS S3.
+
+      If using files was uploaded using a S3Upload block, the `fileList` can be mapped to the `S3Download` block to allow the user to download the files, given that the correct `AwsS3PresignedGetPolicy` is in defined.
```
Comment (JohannMoller) [inline packages/docs/blocks/display/S3Download.yaml:28]: If the files were uploaded using an S3Uplaod block

File: packages/docs/blocks/display/S3Download.yaml [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+              _secret: FILES_S3_ACCESS_KEY_ID
+            secretAccessKey:
+              _secret: FILES_S3_SECRET_ACCESS_KEY
+            region: af-south-1
+            bucket:
+              _secret: FILES_S3_BUCKET
+            write: true
+      ````
+
+      > The examples on this page is not setup with a [`AwsS3Bucket`](/AWSS3) connection, and thus will throw.
```
Comment (JohannMoller) [inline packages/docs/blocks/display/S3Download.yaml:46]: The examples on this page are not set up with an [`AwsS3Bucket`](/AWSS3) connection

**Review by SamTolmay — APPROVED** [review]

**Pull request merged**